scout/server/extensions/bionano_extension.py

```diff
@@ -0,0 +1,203 @@
+
+    def __init__(self):
+        self.url = None
+        self.bionano_username = None
+        self.bionano_password = None
+        self.bionano_token = None
+        self.bionano_user_dict = {}
+
+    def init_app(self, app):
+        self.url = app.config.get("BIONANO_ACCESS")
+        self.bionano_username = app.config.get("BIONANO_USERNAME")
+        self.bionano_password = app.config.get("BIONANO_PASSWORD")
+
+        self._get_token()
+
+    def _get_token(self) -> Optional[Dict]:
+        query = f"{self.url}/administration/api/1.4/login?username={self.bionano_username}&password={self.bionano_password}"
+
+        LOG.info("Authenticating for BioNano Access")
+        json_response = get_request_json(query)
+
+        json_content = json_response.get("content")
+        if not json_content:
+            LOG.debug("No token JSON returned. Message: %s", json_response.get("message"))
+            return None
+        LOG.debug("Token request response was %s", json_content)
+        json_output = json_content.get("output")
+
+        if "user" in json_output:
+            self.bionano_user_dict = json_output.get("user")
+        if "token" in json_output:
+            self.bionano_token = json_output.get("token")
+
+        return json_content
+
+    def _get_auth_cookies(self):
+        """Parse and set authentication cookies based on a prepopulated values of token etc.
+        Called after self._get_token, by methods that need cookies for accessing bionano-access.
+
+        Returns:
+
+        """
+        cookies = {
+            "token": self.bionano_token,
+            "email": self.bionano_user_dict.get("email"),
+            "fullname": self.bionano_user_dict.get("full_name"),
+            "userpk": call_safe(str, self.bionano_user_dict.get("userpk")),
+            "userrole": call_safe(str, self.bionano_user_dict.get("role")),
+        }
+        return cookies
+
+    def _get_json(self, query):
+        json_response = get_request_json(query, cookies=self._get_auth_cookies())
+
+        json_content = json_response.get("content")
+        json_status = json_response.get("status_code")
+        LOG.debug(
+            "Query %s cookies %s status code %s", query, self._get_auth_cookies(), json_status
+        )
+        if not json_content:
+            LOG.debug("No response.")
+            return None
+        LOG.debug("Response was %s", json_content)
+
+        return json_content
+
+    def _post_json(self, query, query_data):
+        json_response = post_data_request_json(query, query_data, cookies=self._get_auth_cookies())
+
+        json_content = json_response.get("content")
+        json_status = json_response.get("status_code")
+        LOG.debug(
+            "Query %s cookies %s status code %s", query, self._get_auth_cookies(), json_status
+        )
+        if not json_content:
+            LOG.debug("No response.")
+            return None
+        LOG.debug("Response was %s", json_content)
```

Comment:
Perhaps remove most of the debug messages in this extension. They can be quite long and clog the LOG:


<img width="1359" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/ff72f0a2-e1f4-4a29-b515-0b2b41260712">

Reply:
Agreed, most were only relevant for actual debug. But DEBUGs shouldn't normally be in the server log though..